### DISCUSSION TITLE
XPath Scraping or RSS Bridge for Complex Sites?

### DISCUSSION THREAD

**Discussion by quella** [FreshRSS/FreshRSS#5700] [Q&A] [answered]
So I have been doing some research on the power of XPath and have not been able to progress on a few websites which currently do not have RSS available.  I have linked a few of the sites I'm trying to scrape below for any thoughts.  Most of the examples on YouTube or even this link here (https://danq.me/2022/09/27/freshrss-xpath/#attachment_20525) cover XPath at a high-level and the sites in the examples are rather easy site and HTML.  

The question is, should I abandon on the XPath route due to site complexity and instead focus my efforts on using something, also listed in the discussions om the forum, RSS Bridge?  I know I'm over my level of expertise, and would like to dive into the best option for complex sites such as these shown below.   

**Sites I'm trying to scrape:**
https://www.artstation.com/?sort_by=latest&dimension=all     [Art site]
https://makerworld.com/de/models/categories?categories=0&orderBy=newUploads     [3d Print Models]

**Comment by Alkarex** [comment, marked as answer]
With the disclaimer that I have only looked at the two sites a few seconds, the first one looks like it is dynamically loaded (so not so appropriate for XPath), while the second looks more appropriate.
In general though, if you can download the wanted page with curl and obtain a straightforward HTML, then the XPath approach is good.
Otherwise, or if there is any additional complexity, an RSS Bridge is most likely better.

**Reply by Frenzie** [reply]
Though with #5079 that wouldn't necessarily be true. ;-)

**Reply by quella** [reply]
The second I have been able to get working with the assistance of @Frenzie in another post.  Working well.  The first I think is deploying some anti-scrapping tech based on some more research.  It seems with the anti-AI movement in the art scene, many sites like this are starting to block all but human-like connections.  Thank you both for your thoughts and support.
